addon/lib/field.js

```diff
@@ -252,7 +256,40 @@ export default Base.extend({
    * @property {*} value
    * @accessor
    */
-  value: reads("answer.value"),
+  value: computed(
+    "question.isCalculated",
+    "calculatedValue",
+    "answer.value",
+    function () {
+      if (this.question.isCalculated) {
+        return this.calculatedValue;
+      }
+
+      return this.answer.value;
+    }
+  ),
+
+  calculatedValue: computed(
+    "document.jexl",
+    "calculatedDependencies.@each.{hidden,value}",
+    "jexlContext",
+    "question.{isCalculated,calcExpression}",
+    function () {
+      if (
+        !this.question.isCalculated ||
+        !this.calculatedDependencies.every(
+          (field) => !field.hidden && typeof field.value === "number"
```

Comment:
Is it really a requirement that calculated fields are only working with numbers? I don't have a concrete use-case, but maybe we'd like to work with strings as well? Would that cause trouble anywhere?

Reply:
Not sure what the backend does here. However, I don't think that string concatenation is a valid use case for a caluma field. We should keep it as simple as possible without allowing stuff that could cause problems if we don't have a use case for it

Reply:
cc @sbor23: do we have this limitation in the backend as well? If we decide to enforce this limitation, I'd prefer to do it alreading (additionally) in `SaveCalculatedFloatQuestion`. I currently don't really see how handling strings as well could cause problems.

Reply:
I still don't think we should support strings since there is no use case for this.. @kaldras what do you think?

Reply:
I think I found the relevant backend code snippet (validators.py):
```py
        question_jexl = jexl.QuestionJexl()
        deps = set(question_jexl.extract_referenced_questions(expr))
        illegal_deps = ", ".join(
            models.Question.objects.filter(
                pk__in=deps, type=models.Question.TYPE_CALCULATED_FLOAT
            ).values_list("slug", flat=True)
        )

        if illegal_deps:
            raise exceptions.ValidationError(
                f"Calc expression references other calculated question: {illegal_deps}"
            )
```

Reply:
The backend doesn't do anything with the input types nor the output types. However we went for a "typed" GraphQL schema for calculated fields. It returns a `FloatAnswer` so we could enforce this more strictly in the future. But theoretically a String could be returned even right now, but I don't know how graphql (frontend or backend) would handle such a case.

Reply:
If I understand correctly the backend would try to evaluate the jexl even for string answers right now but the frontend does not?

I don't see a use case for other types than numbers/integers right now, so I think only accessing number answers would be ok. However it is done, is should be done the same way in the front and backend right?

Reply:
@czosel yes the backend doesn't validate anything except that there are no referenced calculated questions. 

> If I understand correctly the backend would try to evaluate the jexl even for string answers right now but the frontend does not?

For the backend: yes

> I don't see a use case for other types than numbers/integers right now, so I think only accessing number answers would be ok. However it is done, is should be done the same way in the front and backend right?

Probably yeah.

Reply:
We should only allow integer and float questions as valid dependencies since we are talking about a Calculated**Float**Question which has to return a float. Strings don't make any sense.

Reply:
Oh right - I totally missed the fact that it would be quite weird to build a float from strings. @sbor23, can you make the backend validation more strict as @anehx suggested?

Reply:
I'll go ahead and resolve this, because the frontend implementation is good :upside_down_face:

Reply:
> We should only allow integer and float questions as valid dependencies since we are talking about a Calculated**Float**Question which has to return a float. Strings don't make any sense.

This is a wrong assumption IMO.
A valid calcExpression could be something like `'my-text-question'|answer in ['foo', 'bar'] ? 1 : 0`. So `my-text-question` could obviously be a `TextQuestion` (or `StringQuestion`...). So there is no way you can validate the input types without doing actual static analysis on it.

Reply:
Good point, didn't think of the ternary idea.

Reply:
LGTM now